### PR TITLE
(#6)added glossary app integration with cgov platform tests

### LINKED ISSUE #6: Glossary App integration tests
Create integration tests for the following story:
Dictionary App Module in Drupal

**Parent Issue:** 
https://github.com/NCIOCPL/cgov-digital-platform/issues/2694

### Notes

*Some complementary notes if necessary:*
* > Here goes a quote from an email
* Here goes whatever useful information can exist…
* More notes, or technical nuances/direction

### PR DESCRIPTION
**Pull request by kate-mashkina**
Close #6

### CODE REVIEW

**Comment by kate-mashkina** [comment]
@bryanpizzillo  ready for your review

File: cypress/integration/GlossaryAppIntegration/GlossaryAppIntegration.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
+/// <reference types="Cypress" />
+import { Given, When, Then } from 'cypress-cucumber-preprocessor/steps';
+
+Given('user is on home page', () => {
```
Comment (bryanpizzillo) [inline cypress/integration/GlossaryAppIntegration/GlossaryAppIntegration.js:4]: Should this really be here and not in common?

File: cypress/integration/GlossaryAppIntegration/GlossaryAppIntegration.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  cy.visit('/');
+});
+
+Given('user is on Spanish home page', () => {
```
Comment (bryanpizzillo) [inline cypress/integration/GlossaryAppIntegration/GlossaryAppIntegration.js:8]: Same common comment

File: cypress/integration/GlossaryAppIntegration/GlossaryAppIntegration.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+  cy.visit('/espanol');
+});
+
+When('user navigates to Dictionary tab', () => {
```
Comment (bryanpizzillo) [inline cypress/integration/GlossaryAppIntegration/GlossaryAppIntegration.js:12]: I would not do this as part of the dictionaries. This is really the cancer.gov utility bar and not directly related to the dictionary.

File: cypress/integration/GlossaryAppIntegration/GlossaryAppIntegration.js [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+  cy.get("a[href='/publications/dictionaries/cancer-terms']:visible").click();
+});
+
+When('user navigates to Diccionario tab', () => {
```
Comment (bryanpizzillo) [inline cypress/integration/GlossaryAppIntegration/GlossaryAppIntegration.js:16]: Same comment regarding the utility bar

File: cypress/integration/GlossaryAppIntegration/GlossaryAppIntegration.js [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+  cy.get("a[href='/espanol/publicaciones/diccionario']:visible").click();
+});
+
+Then('user is on landing page for Diccionario', () => {
```
Comment (bryanpizzillo) [inline cypress/integration/GlossaryAppIntegration/GlossaryAppIntegration.js:20]: this is really not needed

File: cypress/integration/GlossaryAppIntegration/GlossaryAppIntegration.js [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+  cy.location('pathname').should('be.eq', '/espanol/publicaciones/diccionario');
+});
+
+Then('user is on landing page for the selected Dictionary', () => {
```
Comment (bryanpizzillo) [inline cypress/integration/GlossaryAppIntegration/GlossaryAppIntegration.js:24]: this is really not needed

File: cypress/integration/GlossaryAppIntegration.feature [hunk 7]
```diff
@@ -0,0 +1,64 @@
+Feature: Glossary app is integrated withing CGOV platform
+
+
+    Scenario: User is able to access a default (or “landing”) page that displays all of the expected Dictionary elements for the Dictionary of Cancer terms.
```
Comment (bryanpizzillo) [inline cypress/integration/GlossaryAppIntegration.feature:4]: So this is conflating two different things, you are combining the testing of the cancer.gov utility bar with the ability for the app module to display. These should be two separate scenarios. This scenarion should just have the first when of:
`When the user is viewing the url "/publications/.../dictionary"`

Comment (bryanpizzillo) [reply cypress/integration/GlossaryAppIntegration.feature:4]: The same goes for spanish, but with spanish urls.

File: cypress/integration/GlossaryAppIntegration.feature [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+        Then search options for "Starts with" and "Contains" are displayed
+        And "starts with" is selected by default
+        Then "Browse" option is displayed below search bar
+        And browse option contains all alphabetic letters and "#" appears at the end of the list
+        And each option appears as a link
+        And the page is showing the expand results for letter "A"
+        And the URL does not include "expand=A"
+
+    Scenario: Negative test - user enters a bogus URL (like /chicken at the end of the dictionary URL
+        When user is navigating to bad url "/publications/dictionaries/cancer-terms/chickens" status code is 404
```
Comment (bryanpizzillo) [inline cypress/integration/GlossaryAppIntegration.feature:63]: can this be
```
When user is navigating to bad url "/publications/dictionaries/cancer-terms/chickens"
Then status code is 404
```

File: cypress/integration/GlossaryAppIntegration/GlossaryAppIntegration.js [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+    const paragraph = doc.querySelector('#NCI-glossary-root div p').textContent;
+    expect(paragraph).to.match(/^(?=.*\d).+$/);
+  });
+});
+
+Then('{string} links to {string}', (link, href) => {
+  cy.get(`a[href='${href}']`).should('include.text', link);
+});
+
+Then('user is redirected to 404 error page', () => {
```
Comment (bryanpizzillo) [inline cypress/integration/GlossaryAppIntegration/GlossaryAppIntegration.js:141]: This is probably not needed. There is no redirect to a 404 (a redirect is a HTTP thing and using the term here is confusing). 
```gherkin
Given a user visits the url "/foo/bar"
Then the status is 404
   And the content title is "Page not found"
```
Or something like that.

File: cypress/integration/GlossaryAppIntegration/GlossaryAppIntegration.js [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+
+Then('the following introductory text appears below the page title: {string} and {string}', (beforeCount, afterCount) => {
+  cy.document().then(doc => {
+    const paragraph = doc.querySelector('#NCI-glossary-root div p').textContent;
+    expect(paragraph).to.include(afterCount);
+    expect(paragraph).to.include(beforeCount);
+  });
+});
+
+Then('there is a dynamic number generated by the API that has been inserted', () => {
```
Comment (bryanpizzillo) [inline cypress/integration/GlossaryAppIntegration/GlossaryAppIntegration.js:129]: This text is too vague. Just say `the number of glossary terms appears in the text`. This makes it actually specific to the glossary, which this is, and it makes it clear what you are looking for.

(If we ever add a random number generator to the dictionary, then this text would apply)

File: cypress/integration/GlossaryAppIntegration/GlossaryAppIntegration.js (hunk 1 above)
Comment (kate-mashkina) [reply cypress/integration/GlossaryAppIntegration/GlossaryAppIntegration.js]: gone completely

File: cypress/integration/GlossaryAppIntegration/GlossaryAppIntegration.js (hunk 3 above)
Comment (kate-mashkina) [reply cypress/integration/GlossaryAppIntegration/GlossaryAppIntegration.js]: gone

File: cypress/integration/GlossaryAppIntegration.feature (hunk 8 above)
Comment (kate-mashkina) [reply cypress/integration/GlossaryAppIntegration.feature]: now it is ```
Given a user visits the url "/foo/bar"
Then the status is 404
   And the content title is "Page not found"
```

File: cypress/integration/GlossaryAppIntegration/GlossaryAppIntegration.js (hunk 10 above)
Comment (kate-mashkina) [reply cypress/integration/GlossaryAppIntegration/GlossaryAppIntegration.js]: done

**Pull request merged**